### PR TITLE
Add generic cxx17 and cxx14 support. 

### PR DESCRIPTION
**Pull request by madmongo1**
Also add gcc-cxx17 support without c11 being dragged in (solves build issue in hunter package OpenSSL)

### CODE REVIEW

File: cxx14.cmake [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Copyright (c) 2013-2017, Ruslan Baratov
+# Copyright (c) 2017, Richard Hodges
+# All rights reserved.
+
+if(DEFINED POLLY_CXX14_CMAKE_)
+    return()
```
Comment (ruslo) [inline cxx14.cmake]: Please keep existent style, indentation is 2 here.

File: cxx14.cmake [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+else()
+    set(POLLY_CXX14_CMAKE_ 1)
+endif()
+
+# Don't use polly_init (no generator expected)
```
Comment (ruslo) [inline cxx14.cmake]: There are few requirements for toolchains in Polly, attachment to CMake generator is one of them.

File: cxx17.cmake [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# Copyright (c) 2013-2017, Ruslan Baratov
+# Copyright (c) 2017, Richard Hodges
+# All rights reserved.
+
+if(DEFINED POLLY_CXX17_CMAKE_)
```
Comment (ruslo) [inline cxx17.cmake]: Please add new toolchains to `bin/details/toolchain_table.py`, see CONTRIBUTING:
* https://github.com/ruslo/polly/blob/master/CONTRIBUTING.md

File: gcc-c++17.cmake [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# Copyright (c) 2013, Ruslan Baratov
+# Copyright (c) 2017, Richard Hodges
+# All rights reserved.
+
+if(DEFINED POLLY_GCC_CXX17_CMAKE)
```
Comment (ruslo) [inline gcc-c++17.cmake]: Please change the name of the toolchain to `gcc-cxx17` similar to existent one.

Comment (madmongo1) [reply gcc-c++17.cmake]: I'll revoke this pull request and submit individual ones

File: bin/detail/toolchain_table.py [hunk 5]
```diff
@@ -54,6 +55,7 @@ def verify(self):
 toolchain_table = [
     Toolchain('default', ''),
     Toolchain('cxx11', ''),
+    Toolchain('cxx17', ''),
```
Comment (ruslo) [inline bin/detail/toolchain_table.py]: Already in master, please rebase.

**Comment by ruslo** [comment]
Closing inactive pull request.

**Pull request closed without merging**